nilearn/decomposition/tests/test_canica.py

```diff
@@ -67,5 +90,36 @@ def test_canica_square_img():
     # Smoke test to make sure an error is raised when no data is passed.
     assert_raises(TypeError, canica.fit)
 
+
+def test_component_sign():
+    # BF for issue #570
+
+    # make data (SVD)
+    rng = np.random.RandomState(0)
+    shape = (20, 10, 1)
+    affine = np.eye(4)
+    components = _make_canica_components(shape)
+
+    # make more de-activations than activations
+    for mp in components:
+        mp[rng.randn(*mp.shape) > .8] *= -1.
```

Comment:
In your example you have actually mp.max() == 1 == -mp.min() so the maps don't get the sign flip. If you are wondering, the reason your test pass is that your test is less or equal and it hits the  equal case in both directions.